### PR TITLE
feat: Support wildcards for unauthed paths

### PR DESCRIPTION
**Pull request by purplefox**
### Description 

This PR extends the unauthed paths for AuthenticationPluginHandler to support wildcards. This make it compatible with authentication.skip.paths in rest-utils config.

This is a stacked PR, please just review the top commit.

### Testing done 

Added new test

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/AuthenticationPluginHandler.java [hunk 1]
```diff
@@ -40,25 +43,38 @@
...
     // KsqlAuthorizationProviderHandler
-    unauthenticatedPaths.addAll(KsqlAuthorizationProviderHandler.PATHS_WITHOUT_AUTHORIZATION);
+    final Set<String> unauthenticatedPaths = new HashSet<>(
+        KsqlAuthorizationProviderHandler.PATHS_WITHOUT_AUTHORIZATION);
     // And then we add anything from the property authentication.skip.paths
     // This preserves the behaviour from the previous Jetty based implementation
     final List<String> unauthed = server.getConfig()
         .getList(ApiServerConfig.AUTHENTICATION_SKIP_PATHS_CONFIG);
     unauthenticatedPaths.addAll(unauthed);
+    final StringBuilder pathBuilder = new StringBuilder();
```
Comment (vcrfxia) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/AuthenticationPluginHandler.java:62]: Replace this with `String.join(",", unauthenticatedPaths)`?

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java [hunk 2]
```diff
@@ -95,7 +95,7 @@
   public static final String AUTHENTICATION_SKIP_PATHS_CONFIG = propertyName(
       "authentication.skip.paths");
   public static final String AUTHENTICATION_SKIP_PATHS_DOC =
-      "List of paths where authentication is not required.";
+      "Comma separated of paths where authentication is not required. Wildcards are supported.";
```
Comment (vcrfxia) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java:98]: typo:
```suggestion
      "Comma separated list of paths where authentication is not required. Wildcards are supported.";
```

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/api/AuthTest.java [hunk 3]
```diff
@@ -302,6 +302,14 @@ public void shouldAllowQueryWithSecurityPluginRejectingIfInAuthedPaths() throws
         super::shouldExecutePullQuery, false, false);
   }
 
+  @Test
+  public void shouldAllowQueryWithSecurityPluginRejectingIfInAuthedPathsWithWildcard()
```
Comment (vcrfxia) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/api/AuthTest.java:306]: typo:
```suggestion
  public void shouldAllowQueryWithSecurityPluginRejectingIfInUnauthedPathsWithWildcard()
```

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/AuthenticationPluginHandler.java (hunk 1 above)
Comment (purplefox) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/AuthenticationPluginHandler.java:62]: I missed have missed this method when it came out in Java 8, nice pointer :)

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java (hunk 2 above)
Comment (purplefox) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java:98]: ack

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/api/AuthTest.java (hunk 3 above)
Comment (purplefox) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/api/AuthTest.java:306]: Ack

**Pull request merged**